text/0000-uefi-http-boot.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: UEFI HTTP Boot
+- Start Date: 2016-09-12
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+UEFI supports new kind of boot via HTTP instead of traditional (and slow) TFTP. Implementing this in Foreman is possible with relatively small amount of changes.
+
+# Motivation
+[motivation]: #motivation
+
+Todays firmware is complicated and implementing a HTTP stack is possible, therefore we will see less TFTP in the future. Foreman already supports iPXE HTTP booting which is used for booting virtual machines with iPXE firmware built-in and UEFI HTTP Boot feature will be essentially the same. The feature can enable both UEFI HTTP and iPXE HTTP booting as the implementation requires only two main steps described below.
+
+# Detailed design
+[design]: #detailed-design
+
+In order to support HTTP booting, we need to have an HTTP service deployed. Until now, users were deploying their own services becaue foreman-proxy only provides TFTP service and HTTP API. As part of this RFC, I propose to implement a simple HTTP/HTTPS service as a foreman-proxy plugin serving all files from TFTP directory. Only regular files are supported, listing of directories must be disabled.
```

Comment:
Why not proxy the request to Foreman via "Templates Proxy"-feature but serve the deployed tftp files?